Automattic/newspack-blocks
# feat: merge RAS-ACC work into trunk

### All Submissions:


### Changes proposed in this Pull Request:

This PR merges the epic/ras-acc work into trunk 🎉 

The major changes covered in the Newspack Blocks plugin are related to the modal checkout and checkout button block.

### How to test the changes in this Pull Request:

I'm not sure the best way to test this but I think running smoke tests through a few flows with the other related PRs would be best.

1. Apply this and other PRs related to RAS-ACC (there should be a [plugins](https://github.com/Automattic/newspack-plugin/pull/3582), [newsletters](https://github.com/Automattic/newspack-newsletters/pull/1716), and [theme](https://github.com/Automattic/newspack-theme/pull/2417) one)
2. Smoke test different checkouts with the checkout button block including:
* Memberships
* Subscriptions
* Variations
3. Smoke test the modal checkout from the donate block.
4. Check that existing Checkout Button blocks (ones added when on /trunk) don't throw errors in the editor.
5. Run through anything else you can think of!

### Other information:
